app/views/grade_entry_forms/_form.html.erb

```diff
@@ -89,3 +89,12 @@
                  data: { disable_with: t('working') } %>
   </p>
 <% end %>
+
+<% if @grade_entry_form.id != nil %>
+  <% deletable = @grade_entry_form.count_non_nil == 0 %>
```

Comment:
While I like that you tried to use an existing method, that particular method does a lot more work than necessary. I encourage you to use an association and a query directly; note that `@grade_entry_form` already has a `grades` association, and you can filter out the non-`nil` grades, and use `exists?` to check whether there are any results. You can do this both for this disabling check and for the actual `before_destroy` callback itself (see my above comment).